Podverse/PVDeleter.swift

```diff
@@ -18,25 +18,31 @@ extension Notification.Name {
 class PVDeleter: NSObject {
     
     static func deletePodcast(podcastId: NSManagedObjectID?, feedUrl: String? = nil) {
-        let moc = CoreDataHelper.createMOCForThread(threadType: .privateThread)
-        
-        if let podcastId = podcastId, let podcast = CoreDataHelper.fetchEntityWithID(objectId: podcastId, moc: moc) as? Podcast {
-            podcast.removeFromAutoDownloadList()
-            deleteAllEpisodesFromPodcast(podcast: podcast)
-            moc.delete(podcast)
-            DispatchQueue.main.async {
-                NotificationCenter.default.post(name: .podcastDeleted, object: nil, userInfo: ["feedUrl": podcast.feedUrl as? Any])
+            let moc = CoreDataHelper.createMOCForThread(threadType: .mainThread)
+
+            var podcastToDelete:Podcast!
+            
+            if let podcastId = podcastId, let podcast = CoreDataHelper.fetchEntityWithID(objectId: podcastId, moc: moc) as? Podcast {
+                podcastToDelete = podcast
+            } else if let feedUrl = feedUrl, let podcast = Podcast.podcastForFeedUrl(feedUrlString: feedUrl, managedObjectContext: moc) {
+                podcastToDelete = podcast
             }
-        } else if let feedUrl = feedUrl, let podcast = Podcast.podcastForFeedUrl(feedUrlString: feedUrl, managedObjectContext: moc) {
-            podcast.removeFromAutoDownloadList()
-            deleteAllEpisodesFromPodcast(podcast: podcast)
-            moc.delete(podcast)
-            DispatchQueue.main.async {
-                NotificationCenter.default.post(name: .podcastDeleted, object: nil, userInfo: ["feedUrl": feedUrl as? Any])
+            else {
+                NotificationCenter.default.post(name: .podcastDeleted, 
```

Comment:
Why is this notification post not dispatched to the main thread, but the one in line 43 below is?

If deletePodcast is on the main thread this makes sense to me, since the core data save method calls a completion block.